emails/views.py

```diff
@@ -939,6 +955,66 @@ def _get_email_bytes(
     return (message_content, transport, load_time_s)
 
 
+def _get_developer_mode_action(
+    mask: RelayAddress | DomainAddress,
+) -> DeveloperModeAction | None:
+    """Get the developer mode actions for this mask, if enabled."""
+
+    flag_name = "developer_mode"
+    _, _ = get_waffle_flag_model().objects.get_or_create(
+        name=flag_name,
+        defaults={
+            "note": "MPP-3932: Enable logging and overrides for Relay developers."
+        },
+    )
+
+    if not (
+        flag_is_active_in_task(flag_name, mask.user) and "DEV:" in mask.description
+    ):
+        return None
+
+    if "DEV:simulate_complaint" in mask.description:
+        action = DeveloperModeAction(
+            mask_id=mask.metrics_id,
+            action="simulate_complaint",
+            new_destination_address=f"complaint+{mask.address}@simulator.amazonses.com",
+        )
+    else:
+        action = DeveloperModeAction(mask_id=mask.metrics_id, action="log")
+    return action
+
+
+def _log_dev_notification(
+    log_message: str, dev_action: DeveloperModeAction, notification: dict[str, Any]
+) -> None:
+    """
+    Log notification JSON
+
+    This will log information beyond our privacy policy, so it should only be used on
+    Relay staff accounts with prior permission.
+
+    The notification JSON will be compressed, Ascii85-encoded with padding, and broken
+    into 1024-bytes chunks. This will ensure it fits into GCP's log entry, which has a
+    64KB limit per label value.
+    """
+
+    notification_gza85 = base64.a85encode(
+        zlib.compress(json.dumps(notification).encode()), wrapcol=1024, pad=True
+    ).decode("ascii")
```

Comment:
quesgestion (non-blocking): can this sequence of `base64`, `zlib`, and `json` calls be moved into a helper function `format_notification_for_log` that can help with re-using it exactly the same way each time? E.g., in the test code?